test/e2e/install_test.go

```diff
@@ -0,0 +1,210 @@
+		cmd = fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=%s DD_REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta bash -c \"$(curl -L %s/install_script_agent7.sh)\"", flavor, apiKey, scriptURL)
+		vm.Execute(cmd)
+	} else if mode == "upgrade5" {
+		if flavor != "datadog-agent" {
+			t.Logf("%s not supported on Agent 5", flavor)
+			t.FailNow()
+		}
+		t.Log("Install latest Agent 5")
+		cmd := fmt.Sprintf("DD_API_KEY=%s bash -c \"$(curl -L https://raw.githubusercontent.com/DataDog/dd-agent/master/packaging/datadog-agent/source/install_agent.sh)\"", apiKey)
+		vm.Execute(cmd)
+		t.Log("Install latest Agent 7 RC")
+		cmd = fmt.Sprintf("DD_AGENT_FLAVOR=%s DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=%s DD_UPGRADE=true DD_REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta bash -c \"$(curl -L %s/install_script_agent7.sh)\"", flavor, apiKey, scriptURL)
+		vm.Execute(cmd)
+	}
+
+	t.Run("Check user, config file and service", func(tt *testing.T) {
+		// Check presence of the dd-agent user
+		_, err := vm.ExecuteWithError("id dd-agent")
+		assert.NoError(tt, err, "user datadog-agent does not exist after install")
+		// Check presence of the config file - the file is added by the install script, so this should always be okay
+		// if the install succeeds
+		_, err = vm.ExecuteWithError(fmt.Sprintf("stat /etc/%s/%s", baseName, configFile))
+		assert.NoError(tt, err, fmt.Sprintf("config file /etc/%s/%s does not exist after install", baseName, configFile))
+		// Check presence and ownership of the config and main directories
+		owner := strings.TrimSuffix(vm.Execute(fmt.Sprintf("stat -c \"%%U\" /etc/%s/", baseName)), "\n")
+		assert.Equal(tt, "dd-agent", owner, fmt.Sprintf("dd-agent does not own /etc/%s", baseName))
+		owner = strings.TrimSuffix(vm.Execute(fmt.Sprintf("stat -c \"%%U\" /opt/%s/", baseName)), "\n")
+		assert.Equal(tt, "dd-agent", owner, fmt.Sprintf("dd-agent does not own /opt/%s", baseName))
+		// Check that the service is active
+		if _, err = vm.ExecuteWithError("command -v systemctl"); err == nil {
+			_, err = vm.ExecuteWithError(fmt.Sprintf("systemctl is-active %s", baseName))
+			assert.NoError(tt, err, fmt.Sprintf("%s not running after Agent install", baseName))
+		} else if _, err = vm.ExecuteWithError("command -v initctl"); err == nil {
+			status := strings.TrimSuffix(vm.Execute(fmt.Sprintf("sudo status %s", baseName)), "\n")
+			assert.Contains(tt, "running", status, fmt.Sprintf("%s not running after Agent install", baseName))
+		} else {
+			assert.FailNow(tt, "Unknown service manager")
+		}
+	})
+
+	if flavor == "datadog-agent" {
+		t.Run("Install an extra integration, and create a custom file", func(tt *testing.T) {
+			_, err := vm.ExecuteWithError("sudo -u dd-agent -- datadog-agent integration install -t datadog-bind9==0.1.0")
+			assert.NoError(tt, err, "integration install failed")
+			_ = vm.Execute(fmt.Sprintf("sudo -u dd-agent -- touch /opt/%s/embedded/lib/python3.9/site-packages/testfile", baseName))
```

Comment:
I am curious, why do we do this? To check it is deleted later when removing the agent?

Reply:
We probably test using the integration command, and later test that the custom file is kept after uninstall